service/history/api/consistency_checker.go

```diff
@@ -207,7 +207,7 @@ func (c *WorkflowConsistencyCheckerImpl) getWorkflowContextValidatedByCheck(
 			return nil, err
 		}
 		return NewWorkflowContext(wfContext, release, mutableState), nil
-	case *serviceerror.NotFound:
+	case *serviceerror.NotFound, *serviceerror.NamespaceNotFound:
```

Comment:
this section will never see namespace missing

Reply:
ignore me, just remember mutable state will need to have access to ns cache entry

Reply:
yes, I actually tracked all that calls. It is very unlikely that it will catch it there but still possible.